ecosystem/indexer-grpc/indexer-grpc-table-info/src/internal_indexer_db_service.rs

```diff
@@ -138,11 +144,10 @@ impl InternalIndexerDBService {
         loop {
             let start_time: std::time::Instant = std::time::Instant::now();
             let next_version = self.db_indexer.process_a_batch(start_version)?;
-
             if next_version == start_version {
-                tokio::time::sleep(std::time::Duration::from_millis(100)).await;
-                continue;
+                self.update_receiver.changed().await?;
```

Comment:
add continue back so we don't log garbage

Reply:
use https://docs.rs/tokio/latest/tokio/time/fn.timeout.html just in case the DB stops progressing indefinitely.

Reply:
wrt the `?`, try to think about scenario where either sender or receiver is dropped and if the other side will panic -- we want peaceful quitting.

Reply:
when sender dropped,  receiver will get an error and return. If the receiver is dropped, sender will keep sending updates to channel.